src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,140 @@
+                    }
+                }
+            } else if (parsedInput.equals("todo")) {
+                if (input.replaceAll("\\s", "").length() == 4) {
+                    throw new DukeException(DukeException.Type.EmptyTodo);
+                } else {
+                    Todo t = new Todo(input.substring(5));
+                    System.out.println();
+                    list.add(t);
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     Got it. I've added this task:");
+                    System.out.println("       " + t.toString());
+                    System.out.println(String.format("     Now you have %d tasks in the list.", list.size()));
+                }
+            } else if (parsedInput.equals("event")) {
+                if (input.replaceAll("\\s", "").length() == 5) {
+                    throw new DukeException(DukeException.Type.EmptyEvent);
+                } else {
+                    String[] splitString = input.substring(6).split(" /at ", 2);
+                    Event e = new Event(splitString[0], splitString[1]);
+                    list.add(e);
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     Got it. I've added this task:");
+                    System.out.println("       " + e.toString());
+                    System.out.println(String.format("     Now you have %d tasks in the list.", list.size()));
+                }
+            } else if (parsedInput.equals("deadline")) {
+                if (input.replaceAll("\\s", "").length() == 8) {
+                    throw new DukeException(DukeException.Type.EmptyDeadline);
+                } else {
+                    String[] splitString = input.substring(9).split(" /by ", 2);
+                    Deadline d = new Deadline(splitString[0], splitString[1]);
+                    list.add(d);
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     Got it. I've added this task:");
+                    System.out.println("       " + d.toString());
+                    System.out.println(String.format("     Now you have %d tasks in the list.", list.size()));
+                }
+            } else if (parsedInput.equals("find")) {
+                String query = input.substring(5).toLowerCase();
+                ArrayList<Task> tasksContainingQuery = new ArrayList<Task>();
+                for (int i = 0; i < list.size(); i++) {
+                    Task t = list.get(i);
+                    if (list.get(i).description.toLowerCase().contains(query)) {
+                        tasksContainingQuery.add(t);
+                    }
+                }
+                if (tasksContainingQuery.size() == 0) {
```

Comment:
Logic here involves the task list. It may be more appropriate to extract into a method in a task list class. Ideally UI takes string from user to pass to pass to parser and takes string from duke and prints, since the UI is an interface between duke and user. All other logic can be defined in other classes.